tests/test_merge_transactions.py

```diff
@@ -0,0 +1,188 @@
+from pytest import fixture
+from unittest.mock import patch, Mock, mock_open
+from typing import Generator, List, Optional, Union
+from click.testing import CliRunner
+import pytest
+from personal_finances.bank_interface.nordigen_adapter import (
+    TransactionAmount,
+    NordigenTransaction,
+    NordigenTransactions,
+)
+import json
+from personal_finances.merge_transactions import merge_transactions
+
+
+@fixture(autouse=True)
+def write_json_mock() -> Generator[Mock, None, None]:
+    with patch("personal_finances.merge_transactions.write_json") as mock:
+        yield mock
+
+
+@fixture(autouse=True)
+def listdir_mock() -> Generator[Mock, None, None]:
+    with patch("personal_finances.merge_transactions.os.listdir") as mock:
+        yield mock
+
+
+@fixture(autouse=True)
+def open_mock() -> Generator[Mock, None, None]:
+    m = mock_open()
+    with patch("personal_finances.merge_transactions.open", m):
+        yield m
+
+
+def create_nordigen_transaction(
+    amount: float, id: Optional[str] = None, internal_id: Optional[str] = None
+) -> NordigenTransaction:
+    params: dict = {
+        "bookingDate": "dummy_data",
+        "bookingDatetime": "dummy_data",
+        "transactionAmount": TransactionAmount(currency="EUR", amount=amount),
+    }
+
+    if internal_id is not None:
+        params["internalTransactionId"] = internal_id  # Adjust the key if provided
+
+    if id is not None:
+        params["transactionId"] = id  # Adjust the key if provided
+
+    return NordigenTransaction(**params)
+
+
+TEST_SAME_ID_TRANSACTIONS = [
+    json.dumps(
+        NordigenTransactions(
+            booked=[
+                create_nordigen_transaction(amount=45.00, id="same_id"),
+                create_nordigen_transaction(amount=40.00, internal_id="inter_id"),
+                create_nordigen_transaction(amount=15.00, id="same_id"),
+                create_nordigen_transaction(amount=10.00, internal_id="inter_id"),
+            ],
+            pending=[create_nordigen_transaction(amount=5.00, id="noise")],
+        )
+    ),
+    json.dumps(
+        NordigenTransactions(
+            booked=[
+                create_nordigen_transaction(amount=45.00, id="same_id"),
+                create_nordigen_transaction(amount=40.00, internal_id="inter_id"),
+                create_nordigen_transaction(amount=-15.00, id="same_id"),
```

Comment:
Add a comment saying that an identical `transactionId` represents the same transaction regardless of any other parameter.